test/scDAI.t.sol

```diff
@@ -0,0 +1,446 @@
+import {FixedPointMathLib} from "solmate/utils/FixedPointMathLib.sol";
+import {IPool} from "aave-v3/interfaces/IPool.sol";
+import {IAToken} from "aave-v3/interfaces/IAToken.sol";
+import {IVariableDebtToken} from "aave-v3/interfaces/IVariableDebtToken.sol";
+import {IPoolDataProvider} from "aave-v3/interfaces/IPoolDataProvider.sol";
+import {ERC4626} from "solmate/mixins/ERC4626.sol";
+
+import {Constants as C} from "../src/lib/Constants.sol";
+import {ILendingPool} from "../src/interfaces/aave-v2/ILendingPool.sol";
+import {IProtocolDataProvider} from "../src/interfaces/aave-v2/IProtocolDataProvider.sol";
+import {IAdapter} from "../src/steth/IAdapter.sol";
+import {SparkScDaiAdapter} from "../src/steth/scDai-adapters/SparkScDaiAdapter.sol";
+import {scDAI} from "../src/steth/scDAI.sol";
+
+import {scWETH} from "../src/steth/scWETH.sol";
+import {ISwapRouter} from "../src/interfaces/uniswap/ISwapRouter.sol";
+import {AggregatorV3Interface} from "../src/interfaces/chainlink/AggregatorV3Interface.sol";
+import {PriceConverter} from "../src/steth/PriceConverter.sol";
+import {Swapper} from "../src/steth/Swapper.sol";
+import {IVault} from "../src/interfaces/balancer/IVault.sol";
+import {ICurvePool} from "../src/interfaces/curve/ICurvePool.sol";
+import {ILido} from "../src/interfaces/lido/ILido.sol";
+import {IwstETH} from "../src/interfaces/lido/IwstETH.sol";
+import {IProtocolFeesCollector} from "../src/interfaces/balancer/IProtocolFeesCollector.sol";
+import "../src/errors/scErrors.sol";
+import {FaultyAdapter} from "./mocks/adapters/FaultyAdapter.sol";
+import {Address} from "openzeppelin-contracts/utils/Address.sol";
+
+contract scDAITest is Test {
+    using Address for address;
+    using FixedPointMathLib for uint256;
+
+    event UsdcToEthPriceFeedUpdated(address indexed admin, address newPriceFeed);
+    event ProtocolAdapterAdded(address indexed admin, uint256 adapterId, address adapter);
+    event ProtocolAdapterRemoved(address indexed admin, uint256 adapterId);
+    event NewTargetLtvApplied(address indexed admin, uint256 newTargetLtv);
+    event SlippageToleranceUpdated(address indexed admin, uint256 newSlippageTolerance);
+    event EmergencyExitExecuted(
+        address indexed admin, uint256 wethWithdrawn, uint256 debtRepaid, uint256 collateralReleased
+    );
+    event Reallocated();
+    event Rebalanced(uint256 totalCollateral, uint256 totalDebt, uint256 floatBalance);
+    event ProfitSold(uint256 wethSold, uint256 usdcReceived);
+    event TokenSwapped(address token, uint256 amountSold, uint256 usdcReceived);
+    event Supplied(uint256 adapterId, uint256 amount);
+    event Borrowed(uint256 adapterId, uint256 amount);
+    event Repaid(uint256 adapterId, uint256 amount);
+    event Withdrawn(uint256 adapterId, uint256 amount);
+    event Disinvested(uint256 wethAmount);
+    event RewardsClaimed(uint256 adapterId);
+    event SwapperUpdated(address indexed admin, Swapper newSwapper);
```

Comment:
clean up unused events and import statements

Reply:
done